cmd/github-ldap-user-group-creator/main.go

```diff
@@ -290,6 +302,63 @@ type GroupClusters struct {
 
 var githubRobotIds = sets.New[string]("RH-Cachito", "openshift-bot", "openshift-ci-robot", "openshift-merge-robot")
 
+func deleteInvalidUsers(ctx context.Context, clients map[string]ctrlruntimeclient.Client,
+	kerberosIDs sets.Set[string], ciAdmins sets.Set[string], dryRun bool) error {
+
+	var errs []error
+	for cluster, client := range clients {
+		usersToDelete, err := getUsersWithoutKerberosID(ctx, client, cluster, kerberosIDs)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to get users on cluster %s: %w", cluster, err))
+			continue
+		}
+		for user, identites := range usersToDelete {
+			if ciAdmins.Has(user) {
+				// should never happen
+				logrus.WithField("cluster", cluster).WithField("user", user).Info("Attempt to delete ci-admins user. Skipping...")
+				continue
+			}
+			logrus.WithField("cluster", cluster).WithField("user", user).Info("Deleting user...")
+			if !dryRun {
+				if err := client.Delete(ctx, &userv1.User{ObjectMeta: metav1.ObjectMeta{Name: user}}); err != nil && !errors.IsNotFound(err) {
+					errs = append(errs, fmt.Errorf("failed to delete user %s on cluster %s: %w", user, cluster, err))
+
+				}
+				logrus.WithField("cluster", cluster).WithField("user", user).Info("Deleted successfully!")
```

Comment:
Should be logged ONLY when err==nil
```suggestion
				logrus.WithField("cluster", cluster).WithField("user", user).Info("Deleted successfully.")
```